src/site/content/en/blog/google-search-sw/index.md

```diff
@@ -0,0 +1,463 @@
+Information about experiments that might affect the service worker's behavior or
+network requests in general can be included directly in this customized service
+worker script, and changing the sets of experiments that are active for a user
+can be done via a combination of traditional techniques, like browser cookies,
+as well as serving updated code in the registered service worker URL.
+
+Using a dynamically generated service worker script also makes it easier to
+provide an escape hatch in the unlikely event that a service worker
+implementation has a fatal bug that needs to be avoided. The folks responsible
+for the Google Search service worker can have the dynamic service worker
+response include a [no-op implementation](https://stackoverflow.com/a/38980776/385997),
+effectively disabling the service worker for some or all of the current users.
+
+### Coordinating updates
+
+One of the toughest challenges facing any real-world service worker deployment
+is to devise a reasonable tradeoff between avoiding the network in favor of the
+cache, while at the same time, ensuring that existing users get critical updates
+and changes soon after they're deployed to production. The right balance depends
+on a lot of factors:
+
+* Whether your web app is a long-lived [single page app](https://en.wikipedia.org/wiki/Single-page_application)
+  that a user keeps open indefinitely, without navigating to new pages.
+* What the deployment cadence is for updates to your backend web server.
+* Whether the average user would tolerate using a slightly out-of-date version
+  of your web app, or whether freshness is the top priority.
+
+While experimenting with service workers, the Google Search team made sure to
+keep the experiments running across a number of scheduled backend updates, to
+ensure that the metrics and user experience would more closely match what return
+users would end up seeing in the real-world.
+
+{% Aside 'caution' %}
+It's important to remember that shipping a service worker
+is **not a one-time deployment**—you need to have a process in place, tailored to
+your own production infrastructure, to make sure that updates happen smoothly
+over time!
+{% endAside %}
+
+**Solution**
+
+After testing a number of different configuration options, the Google Search
+team found that the following setup provided the right balance between freshness
+and cache-utilization.
+
+The service worker script URL is served with the
+`Cache-Control: private, max-age=1500` (1500 seconds, or 25 minutes) response
+header, and it's
+[registered with updateViaCache set to 'all'](https://developers.google.com/web/updates/2018/06/fresher-sw#updateviacache)
+to ensure that the header is honored. The Google Search web backend is, as you
+might imagine, a large, globally distributed set of servers that requires as
+close to 100% uptime as possible. Deploying a change that would affect the
+service worker script's contents is done in a rolling fashion. If a user hits a
```

Comment:
This is a giant paragraph. Might be nice to break it up a bit.

Maybe new paragraph at 'If' and another new paragraph at 'Additionally'?